server/src/main/java/org/apache/druid/client/QueryableCoordinatorServerView.java

```diff
@@ -0,0 +1,140 @@
+import com.google.common.collect.Iterables;
+import com.google.inject.Inject;
+import org.apache.druid.client.selector.ServerSelector;
+import org.apache.druid.client.selector.TierSelectorStrategy;
+import org.apache.druid.guice.ManageLifecycle;
+import org.apache.druid.guice.annotations.EscalatedClient;
+import org.apache.druid.guice.annotations.Smile;
+import org.apache.druid.java.util.emitter.service.ServiceEmitter;
+import org.apache.druid.java.util.http.client.HttpClient;
+import org.apache.druid.query.DataSource;
+import org.apache.druid.query.QueryToolChestWarehouse;
+import org.apache.druid.query.QueryWatcher;
+import org.apache.druid.timeline.DataSegment;
+import org.apache.druid.timeline.SegmentId;
+import org.apache.druid.timeline.VersionedIntervalTimeline;
+import org.apache.druid.utils.CollectionUtils;
+
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Map;
+
+/**
+ * ServerView of coordinator for the state of segments being loaded in the cluster.
+ *
+ * <p>This class extends {@link BrokerServerView} and implements {@link CoordinatorTimeline}.
+ * The main distinction between this class and {@link CoordinatorServerView} is the maintenance of a timeline
+ * of {@link ServerSelector} objects, while the other class stores {@link SegmentLoadInfo} object in its timeline.</p>
+ *
+ * <p>A new timeline class (implementing {@link TimelineServerView}) is required for
+ * {@link org.apache.druid.segment.metadata.CoordinatorSegmentMetadataCache}, which will run on the Coordinator.</p>
+ */
+@ManageLifecycle
+public class QueryableCoordinatorServerView extends BrokerServerView implements CoordinatorTimeline
+{
+  private final FilteredServerInventoryView baseView;
+
+  @Inject
+  public QueryableCoordinatorServerView(
+      final QueryToolChestWarehouse warehouse,
+      final QueryWatcher queryWatcher,
+      final @Smile ObjectMapper smileMapper,
+      final @EscalatedClient HttpClient httpClient,
+      FilteredServerInventoryView baseView,
+      TierSelectorStrategy tierSelectorStrategy,
+      ServiceEmitter emitter,
+      CoordinatorSegmentWatcherConfig segmentWatcherConfig
+  )
+  {
+    super(warehouse, queryWatcher, smileMapper, httpClient, baseView, tierSelectorStrategy, emitter, new BrokerSegmentWatcherConfig() {
+      @Override
+      public boolean isAwaitInitializationOnStart()
+      {
+        return segmentWatcherConfig.isAwaitInitializationOnStart();
+      }
+    });
+    this.baseView = baseView;
+  }
+
+  /**
+   * Since this class maintains a timeline of {@link ServerSelector} objects,
+   * this method converts and returns a new timeline of the object {@link SegmentLoadInfo}.
+   *
+   * @param dataSource dataSoruce
+   * @return timeline for the given dataSource
+   */
+  @Override
+  public VersionedIntervalTimeline<String, SegmentLoadInfo> getTimeline(DataSource dataSource)
+  {
+    String table = Iterables.getOnlyElement(dataSource.getTableNames());
```

Comment:
NIt: Lets use `CollectionUtils.getOnlyElement()` so that a better error message is thrown to the user.